mantidproject/mantid
# Indirect - Chi squared in Result workspace and plot combobox

**Description of work.**
This PR ensures that Chi_squared is added to the bottom row of the Results workspaces for Indirect Data Analysis (MSDFit, IqtFit, ConvFit and F(Q) Fit).
This PR also allows the plotting of Chi_squared in the output options.

**To test:**
1.`Interfaces`->`Indirect`->`Data Analysis`
2. In turn, run a fit on MSDFit, IqtFit, ConvFit and F(Q) Fit using the data below
3. The _Result workspace's last row should be Chi-Squared
4. An option to plot chi-squared should be given in the output options in the `Plot` Combobox
5. This should plot the values that can be found in the result workspace

**Data**
*MSDFit:*
[osi104367-104371_graphite002_red_elwin_eq.zip](https://github.com/mantidproject/mantid/files/2910130/osi104367-104371_graphite002_red_elwin_eq.zip)

*IqtFit:*
[irs26176_graphite002_iqt.zip](https://github.com/mantidproject/mantid/files/2910139/irs26176_graphite002_iqt.zip)

*ConvFit:*
[irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2910143/irs26176_graphite002_red.zip)
[irs26173_graphite002_res.zip](https://github.com/mantidproject/mantid/files/2910144/irs26173_graphite002_res.zip)

*FQFit:*
[IN16B_125878_QLd_Result.zip](https://github.com/mantidproject/mantid/files/2910147/IN16B_125878_QLd_Result.zip)

The plotting options should be disbaled for F(Q) Fit because there is only one data point. Look in the `IN16B_125878_QLd_Result_HWHM_FofQFit_ChudleyElliot` workspace for chi_squared.

Fixes #24111 

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
